### PR TITLE
Another attempt to fix the run start problem

### PR DESCRIPTION
**Pull request by darrylmasson**
Reworks how the detector command acknowledgement is accessed, in the hopes that we can finally solve the issue of rundocs going missing because we don't find the start ack time.

### CODE REVIEW

File: dispatcher/MongoConnect.py [hunk 1]
```diff
@@ -393,15 +396,14 @@ def set_stop_time(self, number, detectors, force):
             return
         try:
             time.sleep(0.5) # this number depends on the CC command polling time
-            endtime = self.get_ack_time(detectors, 'stop')
-            if endtime is None:
+            if (endtime := self.get_ack_time(detectors, 'stop') )is None:
```
Comment (petergaemers) [inline dispatcher/MongoConnect.py:399]: I know this is a walrus, but what does it actually do in this case?

File: dispatcher/MongoConnect.py [hunk 2]
```diff
@@ -481,18 +489,19 @@ def process_commands(self):
...
                 else:
-                    dt = (next_cmd['createdAt'] - datetime.datetime.utcnow()).total_seconds()
+                    dt = (next_cmd['createdAt'] - now()).total_seconds()
                 if dt < 0.01:
                     oid = next_cmd.pop('_id')
-                    ret = self.collections['outgoing_commands'].insert_one(next_cmd)
-                    self.collections['command_queue'].delete_one({'_id': oid})
-                    self.command_oid[next_cmd['detector']][next_cmd['command']] = ret.inserted_id
+                    ret = outgoing.insert_one(next_cmd)
+                    self.incoming.delete_one({'_id': oid})
```
Comment (petergaemers) [inline dispatcher/MongoConnect.py:504]: Should this self be there? This won't refer to line 493

File: dispatcher/MongoConnect.py (hunk 1 above)
Comment (petergaemers) [reply dispatcher/MongoConnect.py:399]: Nevermind, got it

**Review by petergaemers — COMMENTED** [review]
I'm just not sure about the self in 504

File: dispatcher/MongoConnect.py (hunk 2 above)
Comment (darrylmasson) [reply dispatcher/MongoConnect.py:504]: Good catch

File: dispatcher/MongoConnect.py (hunk 1 above)
Comment (darrylmasson) [reply dispatcher/MongoConnect.py:399]: There's a space missing there anyway

File: dispatcher/MongoConnect.py [hunk 3]
```diff
@@ -481,18 +489,19 @@ def process_commands(self):
...
                     dt = 10
                 else:
-                    dt = (next_cmd['createdAt'] - datetime.datetime.utcnow()).total_seconds()
+                    dt = (next_cmd['createdAt'] - now()).total_seconds()
                 if dt < 0.01:
                     oid = next_cmd.pop('_id')
-                    ret = self.collections['outgoing_commands'].insert_one(next_cmd)
-                    self.collections['command_queue'].delete_one({'_id': oid})
-                    self.command_oid[next_cmd['detector']][next_cmd['command']] = ret.inserted_id
+                    ret = outgoing.insert_one(next_cmd)
```
Comment (RoBGlaBe) [inline dispatcher/MongoConnect.py:503]: ret holds the ID not the document including next_cmd. That's what you want, I guess?

File: dispatcher/MongoConnect.py (hunk 2 above)
Comment (RoBGlaBe) [inline dispatcher/MongoConnect.py]: "self." is probably not what you want, as you defined it just before while. Or is there a self.incoming, too?

Comment (darrylmasson) [reply dispatcher/MongoConnect.py]: Peter also noticed this one. `self` isn't supposed to be there

**Review by petergaemers — APPROVED** [review]

**Pull request merged**